python/cuml/svm/svc.pyx

```diff
@@ -111,6 +112,83 @@ cdef extern from "cuml/svm/svc.hpp" namespace "ML::SVM" nogil:
         math_t *preds, math_t buffer_size, bool predict_class) except +
 
 
+def apply_class_weight(handle, sample_weight, class_weight, y, verbose, output_type, dtype) -> CumlArray:
+    """
+    Scale the sample weights with the class weights.
+
+    Returns the modified sample weights, or None if neither class weights
+    nor sample weights are defined. The returned weights are defined as
+
+    sample_weight[i] = class_weight[y[i]] * sample_weight[i].
+
+    Parameters:
+    -----------
+    handle : cuml.Handle
+        Specifies the cuml.handle that holds internal CUDA state for
+        computations in this model. 
+    sample_weight: array-like (device or host), shape = (n_samples, 1)
+        sample weights or None if not given
+    class_weight : dict or string (default=None)
+        Weights to modify the parameter C for class i to class_weight[i]*C. The
+        string 'balanced' is also accepted, in which case ``class_weight[i] =
+        n_samples / (n_classes * n_samples_of_class[i])``
+    y: array of floats or doubles, shape = (n_samples, 1)
+    verbose : int or boolean, default=False
+        Sets logging level. It must be one of `cuml.common.logger.level_*`.
+        See :ref:`verbosity-levels` for more info.
+    output_type : {{'input', 'array', 'dataframe', 'series', 'df_obj', \
+        'numba', 'cupy', 'numpy', 'cudf', 'pandas'}}, default=None
+        Return results and set estimator attributes to the indicated output
+        type. If None, the output type set at the module level
+        (`cuml.global_settings.output_type`) will be used. See
+        :ref:`output-data-type-configuration` for more info.
+    dtype : dtype for sample_weights
+
+    Returns
+    --------
+    sample_weight: device array shape = (n_samples, 1) or None
+    """
+    if class_weight is None:
+        return sample_weight
+
+    if y is CumlArray:
+        y_m = y
+    else:
+        y_m, _, _, _ = input_to_cuml_array(y, check_cols=1)
+
+    le = LabelEncoder(handle=handle,
+                      verbose=verbose,
+                      output_type=output_type)
+    labels = y_m.to_output(output_type='series')
+    encoded_labels = cp.asarray(le.fit_transform(labels))
+    n_samples = y_m.shape[0]
```

Comment:
I recognize that this was copied from the previous implementation, but maybe something we can check. It seems that we are dropping semantic information via the `cp.asarray()` transformation, no? So why convert to a `series` instead of just to an array in the line before?

Reply:
Got it, thanks for checking!